syft/frameworks/torch/hook/hook.py

```diff
@@ -1169,37 +1169,46 @@ def clip_grad_norm_remote_(parameters, max_norm, norm_type=2):
             Returns:
                 Total norm of the parameters (viewed as a single vector).
             """
+
+            def param_is_pointer_tensor(param):
+                """
+                A list of parameters is remote if all params contained in the list are
+                remote (i.e., the child of each param is a pointer tensor).
+                This method checks if a single param is indeed remote, so whether
+                the child of a parameter is a pointer tensor
+                """
+                return hasattr(param, "child") and isinstance(param.child, PointerTensor)
+
             if isinstance(parameters, torch.Tensor):
-                # Remote PySyft tensor
-                if hasattr(parameters, "child") and isinstance(
-                    parameters.child, syft.generic.pointers.pointer_tensor.PointerTensor
-                ):
-                    worker = parameters.location
                 parameters = [parameters]
+
             parameters = list(filter(lambda p: p.grad is not None, parameters))
             max_norm = float(max_norm)
             norm_type = float(norm_type)
             if norm_type == inf:
                 total_norm = max(p.grad.data.abs().max() for p in parameters)
             else:
-                # Remote PySyft tensor
-                if hasattr(parameters, "child") and isinstance(
-                    parameters.child, syft.generic.pointers.pointer_tensor.PointerTensor
-                ):
+                # all parameters are remote
+                if all([param_is_pointer_tensor(param) for param in parameters]):
                     total_norm = torch.zeros(1)
-                    # Let's send the total norm over to the remote worker where the remote tensor is
-                    total_norm = total_norm.send(worker)
-                # Local PyTorch tensor
+                    # Let's send the total norm over to the remote where the remote tensor is
+                    total_norm = total_norm.send(parameters[0].location)
```

Comment:
why is this only sent to one location

Reply:
Because the location of all parameters in one model is the same. So, I could have picked the location any random parameter within the model.